packages/pattern-picker/src/use-query-patterns.ts

```diff
@@ -3,6 +3,28 @@ import { useQuery, UseQueryResult } from 'react-query';
 import wpcomRequest from 'wpcom-proxy-request';
 import type { Pattern } from './types';
 
+/**
+ * This query isn't really a query, it's just meant to use Calypso's react-query async storage to store the order for the user forever
+ *
+ * @param length the length of the array to order
+ * @returns an array of numbers in a random order
+ */
+function useOrder( length: number ) {
+	return useQuery(
+		[ 'ptk/order', length ],
+		async () => {
+			const indices = Array.from( { length }, ( _, i ) => i );
+			const firstFour = indices.slice( 0, 4 ).sort( () => Math.random() - 0.5 );
```

Comment:
Works great. Some optional feedback. Returning -1, 0 or 1 for sorting seems more readable to me
```suggestion
			const firstFour = indices.slice( 0, 4 ).sort( () => Math.floor(Math.random() * 3) - 1 );
```

Reply:
This is a pretty common way to shuffle an array.